KlayGE/Core/Src/Kernel/ResLoader.cpp

```diff
@@ -141,6 +143,18 @@ namespace KlayGE
 			exe_path_ = "/data/data/" + exe_path_;
 #endif
 		}
+#elif defined KLAYGE_PLATFORM_DARWIN
+		char buffer[1024];
+		uint32_t size = sizeof(buffer);
+		if (_NSGetExecutablePath(buffer, &size) == 0)
+		{
+			exe_path_ = buffer;
+			exe_path_ = exe_path_.substr(0, exe_path_.find_last_of("/") + 1);
+		}
+		else
+		{
+			KlayGE::LogWarn("buffer too small for executablePath");

```

Comment:
Is there any way we can avoid this? Like calling _NSGetExecutablePath(nullptr, &size) to get the size first, then resize the vector or string.